pennylane_lightning/lightning_qubit/lightning_qubit2.py

```diff
@@ -0,0 +1,509 @@
+        program.add_transform(decompose, stopping_condition=stopping_condition, name=self.name)
+        program.add_transform(qml.transforms.broadcast_expand)
+
+        if config.gradient_method == "adjoint":
+            _add_adjoint_transforms(program)
+        return program, config
+
+    # pylint: disable=unused-argument
+    def execute(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ) -> Result_or_ResultBatch:
+        """Execute a circuit or a batch of circuits and turn it into results.
+
+        Args:
+            circuits (Union[QuantumTape, Sequence[QuantumTape]]): the quantum circuits to be executed
+            execution_config (ExecutionConfig): a datastructure with additional information required for execution
+
+        Returns:
+            TensorLike, tuple[TensorLike], tuple[tuple[TensorLike]]: A numeric result of the computation.
+        """
+        results = []
+        for circuit in circuits:
+            circuit = circuit.map_to_standard_wires()
+            results.append(simulate(circuit, self._statevector))
+
+        return tuple(results)
+
+    def supports_derivatives(
+        self,
+        execution_config: Optional[ExecutionConfig] = None,
+        circuit: Optional[qml.tape.QuantumTape] = None,
+    ) -> bool:
+        """Check whether or not derivatives are available for a given configuration and circuit.
+
+        ``LightningQubit2`` supports adjoint differentiation with analytic results.
+
+        Args:
+            execution_config (ExecutionConfig): The configuration of the desired derivative calculation
+            circuit (QuantumTape): An optional circuit to check derivatives support for.
+
+        Returns:
+            Bool: Whether or not a derivative can be calculated provided the given information
+
+        """
+        if execution_config is None and circuit is None:
+            return True
+        if execution_config.gradient_method not in {"adjoint", "best"}:
+            return False
+        if circuit is None:
+            return True
+        return _supports_adjoint(circuit=circuit)
+
+    def compute_derivatives(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ):
+        """Calculate the jacobian of either a single or a batch of circuits on the device.
+
+        Args:
+            circuits (Union[QuantumTape, Sequence[QuantumTape]]): the circuits to calculate derivatives for
+            execution_config (ExecutionConfig): a datastructure with all additional information required for execution
+
+        Returns:
+            Tuple: The jacobian for each trainable parameter
+        """
+        batch_obs = execution_config.device_options.get("batch_obs", self._batch_obs)
+        return tuple(
+            jacobian(circuit, self._statevector, batch_obs=batch_obs) for circuit in circuits
+        )
```

Comment:
Should we have a check for gradient_method here?

Reply:
This is a preprocessing step handled by dev.preprocess, which is called in qml.execute. It's stated in the documentation that using dev.execute/compute_derivatives is not guaranteed to work with tapes that are not preprocessed.

Reply:
To clarify, I am wondering if the preprocessing step ensures that `qml.execute` only calls `computing_derivatives` when using the adjoint Jacobian method, or if it will also call it if the gradient method is valid for the device. Additionally, what happens if there are two native methods for gradient calculation?